Control/Monad/Indexed/Cont.hs

```diff
@@ -52,6 +52,24 @@ instance Monad m => IxMonadCont (IxContT m) where
 	reset e = IxContT $ \k -> runIxContT e return >>= k
 	shift e = IxContT $ \k -> e (\a -> IxContT (\k' -> k a >>= k')) `runIxContT` return
 
+callCC :: Monad m => ((forall i b. a -> IxContT m o i b) -> IxContT m r o a) -> IxContT m r o a
+callCC f = shift (\k -> f (adapt k) >>>= k)
+	where
+		-- Both 'shift' and 'callCC' capture the current continuation up to the
```

Comment:
Mostly my slowness, but is there an equation we can give to make the statement "callCC continuations never return" precise? I think perhaps the following?

callCC (\k -> f (\a -> k a >>>= g))   =    callCC (\k -> f (\a -> k a >>>= return undefined))